packages/intent-filler/src/strategies/basic.ts

```diff
@@ -0,0 +1,441 @@
+import { encodePacked, keccak256, toHex } from "viem"
+import { ADDRESS_ZERO, fetchTokenUsdPriceOnchain, getOrderCommitment } from "@/utils"
+import { INTENT_GATEWAY_ABI } from "@/config/abis/IntentGateway"
+import { ERC20_ABI } from "@/config/abis/ERC20"
+import { addresses, assets } from "@/config/chain"
+
+export class BasicFiller implements FillerStrategy {
+	name = "BasicFiller"
+
+	private wallet: ethers.Wallet
+	private provider: ethers.providers.Provider
+	private contract: ethers.Contract
+	private chain: string
+
+	constructor(chain: string, privateKey: string, provider: ethers.providers.Provider, contractAddress: string) {
+		this.chain = toHex(chain)
+		this.provider = provider
+		this.wallet = new ethers.Wallet(privateKey, provider)
+		this.contract = new ethers.Contract(contractAddress, INTENT_GATEWAY_ABI, this.wallet)
+	}
+
+	/**
+	 * Determines if this strategy can fill the given order
+	 * @param order The order to check
+	 * @param config The filler configuration
+	 * @returns True if the strategy can fill the order
+	 */
+	async canFill(order: Order, config: FillerConfig): Promise<boolean> {
+		try {
+			const currentChain = this.chain
+			const destChain = toHex(order.destChain)
+
+			if (!destChain.includes(currentChain)) {
+				console.debug(`Order destined for chain ${destChain}, we're on ${currentChain}`)
+				return false
+			}
+
+			const currentBlock = await this.provider.getBlockNumber()
+			const deadline = BigInt(order.deadline)
+
+			if (deadline < currentBlock) {
+				console.debug(`Order expired at block ${deadline}, current block ${currentBlock}`)
+				return false
+			}
+
+			const isAlreadyFilled = await this.checkIfOrderFilled(order)
+			if (isAlreadyFilled) {
+				console.debug(`Order is already filled`)
+				return false
+			}
+
+			const hasEnoughTokens = await this.checkTokenBalances(order.outputs)
+			if (!hasEnoughTokens) {
+				console.debug(`Insufficient token balances for order`)
+				return false
+			}
+
+			const orderValue = this.calculateOrderValue(order)
+			const requiredConfirmations = config.confirmationPolicy.getConfirmationBlocks(
+				this.chain,
+				orderValue.toString(),
+			)
+
+			// If the order is close to expiry and we need many confirmations, we might not want to fill
+			const blocksRemaining = Number(deadline - BigInt(currentBlock))
+			if (blocksRemaining < requiredConfirmations * 2) {
+				console.debug(
+					`Order too close to expiry for comfort: ${blocksRemaining} blocks remaining, ${requiredConfirmations} confirmations required`,
+				)
+				return false
+			}
+
+			return true
+		} catch (error) {
+			console.error(`Error in canFill:`, error)
+			return false
+		}
+	}
+
+	/**
+	 * Calculates the expected profitability of filling this order
+	 * @param order The order to calculate profitability for
+	 * @returns The expected profit in a normalized unit (usually USD value or ETH equivalent)
+	 */
+	async calculateProfitability(order: Order): Promise<number> {
```

Comment:
For the basic filler, we probably allow all orders without any profitability checks.

Reply:
Okay I will remove this or make it optional